### PR TITLE
p004-style-markup

### PR DESCRIPTION
**Pull request by spideyinf**
E hỏi Duy về việc break các changes nhỏ ra rồi chị nhé, công nhận lần này nhiều changes thật lần tới e sẽ rút kinh nghiệm

### CODE REVIEW

File: source/fonts/demo.css [hunk 1]
```diff
@@ -0,0 +1,370 @@
+*{margin: 0;padding: 0;list-style: none;}
+/*
+KISSY CSS Reset
+理念：1. reset 的目的不是清除浏览器的默认样式，这仅是部分工作。清除和重置是紧密不可分的。
+2. reset 的目的不是让默认样式在所有浏览器下一致，而是减少默认样式有可能带来的问题。
```
Comment (emily0604) [inline source/fonts/demo.css]: what's this file

File: source/modules/elements/elements.pug [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+      )
+
+
+
+//-
+//- Elements :: Textarea
+//-
+mixin textarea(name, placeholder, label)
+  .form-group
+    if lable !== ''
```
Comment (emily0604) [inline source/modules/elements/elements.pug]: lable ?

File: source/modules/elements/elements.pug [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+      placeholder= placeholder
+    )&attributes(attributes)
+
+
+
+//-
+//- Elements :: Image mixins
+//-
+
+mixin img-responsive(_src,_alt)
```
Comment (emily0604) [inline source/modules/elements/elements.pug]: why using _

File: source/modules/sidebar/sidebar.pug [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  .sidebar
+    h2.sidebar__user-avatar
+        a(href="#") User avatar image
+    p.sidebar__username Emily Doe
+    ul.nav.nav-stacked.sidebar__nav
+      li.nav-item.active
+        a.nav-link(href='#')
+          i.anticon.icon-user
+          span.text User profile
+      li.nav-item
```
Comment (emily0604) [inline source/modules/sidebar/sidebar.pug]: please using data json. after that using loop to print it

File: source/modules/style-guide/style-guide.pug [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      span.style-guide__project-name= projectName
+      |. The template includes colours, typography, icons, buttons, forms…
+
+    ol.style-guide__list
+      li.style-guide__list-item
+        span.guide__title Changelog
+        block
+        div.changelog
+          - blocks.changelog()
+      li.style-guide__list-item
```
Comment (emily0604) [inline source/modules/style-guide/style-guide.pug]: please using data json. after that using loop

File: source/modules/style-guide/style-guide.pug [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+mixin colorPalette(color, hexCode)
+  div.card
+    div.card-img-top(style=`background-color: ${hexCode}; height: 10rem;`)
+    div.card-body
+      h5.card-title= color
+      p.card-text= hexCode
+
+mixin typoGuide(fontFamily, fontSize, fontWeight, color)
+    ul.my-2.guide-content__typo
+      li.font-weight-bold Font-family:
```
Comment (emily0604) [inline source/modules/style-guide/style-guide.pug]: please using data json. after that using loop

File: source/p004-style-guide.pug [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    +head(head.p004-style-guide)
+
+  body
+  +style-guide('Fram.io - Demo project', '1.1', '05/10/2017')
+
+    +set-block('changelog')
+      ul.guide-content__changelog
+        li.font-weight-normal 05/10/2017 - V1.1
+          ul.guide-content__changelog__detail
+           li Change font to Muli (Google font support)
```
Comment (emily0604) [inline source/p004-style-guide.pug]: please using data json. after that using loop

File: source/sass/base/_global.scss [hunk 8]
```diff
@@ -0,0 +1,6 @@
+// Global changes
+//////////////////////////////////////////////////////////////////
+
+body {
+  background-color: $gray-100;
```
Comment (emily0604) [inline source/sass/base/_global.scss:5]: are you sure. could use using customizing of bootstrap ?

File: source/sass/base/_typography.scss [hunk 9]
```diff
@@ -0,0 +1,28 @@
+body {
+  font-weight: $font-weight-normal;
```
Comment (emily0604) [inline source/sass/base/_typography.scss:2]: check bootstrap variable ?

File: source/sass/components/_ui-elements.scss [hunk 10]
```diff
@@ -0,0 +1,18 @@
+
+.fram-welcome {
+  width: 100%;
+  height: 100vh;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  .fram-welcome__logo {
```
Comment (emily0604) [inline source/sass/components/_ui-elements.scss]: using sass to make sorter way to write fram-welcome__logo

File: source/sass/pages/style-guide.scss [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+      }
+    }
+  }
+
+.guide-content__changelog {
+  padding-left: 3.25rem;
+
+}
+
+  .guide-content__changelog__detail {
```
Comment (emily0604) [inline source/sass/pages/style-guide.scss]: Using sass to write make writing sorter

**Review by emily0604 — CHANGES_REQUESTED** [review]

File: source/fonts/demo.css (hunk 1 above)
Comment (spideyinf) [reply source/fonts/demo.css]: I will remove this. This is a file from anticon font package.

File: source/modules/elements/elements.pug (hunk 2 above)
Comment (spideyinf) [reply source/modules/elements/elements.pug]: Will fix this

File: source/modules/elements/elements.pug (hunk 3 above)
Comment (spideyinf) [reply source/modules/elements/elements.pug]: Will fix this

File: source/fonts/demo.css (hunk 1 above)
Comment (spideyinf) [reply source/fonts/demo.css]: Deleted

File: source/modules/elements/elements.pug (hunk 2 above)
Comment (spideyinf) [reply source/modules/elements/elements.pug]: Fixed

File: source/modules/sidebar/sidebar.pug (hunk 4 above)
Comment (spideyinf) [reply source/modules/sidebar/sidebar.pug]: I will update this with Data.json in page p002-p003-user-and-settings (next pull)

File: source/modules/style-guide/style-guide.pug (hunk 6 above)
Comment (spideyinf) [reply source/modules/style-guide/style-guide.pug]: Updated with data.json and loop

File: source/p004-style-guide.pug (hunk 7 above)
Comment (spideyinf) [reply source/p004-style-guide.pug]: Updated with data.json and loop

File: source/sass/base/_global.scss (hunk 8 above)
Comment (spideyinf) [reply source/sass/base/_global.scss:5]: This is a bootstrap var.

File: source/sass/base/_typography.scss (hunk 9 above)
Comment (spideyinf) [reply source/sass/base/_typography.scss:2]: It is ok

File: source/sass/components/_ui-elements.scss (hunk 10 above)
Comment (spideyinf) [reply source/sass/components/_ui-elements.scss]: Using sass nesting selector, but consider refer to this article: https://www.sitepoint.com/beware-selector-nesting-sass/

File: source/modules/sidebar/sidebar.pug (hunk 4 above)
Comment (emily0604) [reply source/modules/sidebar/sidebar.pug]: please add for this pull. pack it !

File: gulpfile.js [hunk 12]
```diff
@@ -79,7 +79,7 @@ var scss = {
     precison: 3,
     errLogToConsole: true,
     includePaths: [
-      './node_modules/bootstrap-sass/assets/stylesheets',
+      // './node_modules/bootstrap/assets/stylesheets',
```
Comment (emily0604) [inline gulpfile.js]: remove this if you dont need

File: source/modules/sidebar/sidebar.pug [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+    h2.sidebar__user-avatar
+        a(href="#") User avatar image
+    p.sidebar__username Emily Doe
+    ul.nav.nav-stacked.sidebar__nav
+      each item in data
+        li.nav-item.active
+          a.nav-link(href='#')
+            i(class=`anticon icon-${item.icon}`)
+            span.text= item.title
+    h4.sidebar__footer
```
Comment (emily0604) [inline source/modules/sidebar/sidebar.pug]: please check indent

File: source/modules/style-guide/style-guide.pug [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+
+
+mixin iconsGuide(iconfontName, source)
+  p
+  | The icons are from
+  span.font-weight-bold #{iconfontName}
+  |  and can be downloaded from
+  a(href= source).font-weight-normal #{source}
+
+//- mixin buttonsGuide()
```
Comment (emily0604) [inline source/modules/style-guide/style-guide.pug]: remove all comment if you dont use

**Review by emily0604 — CHANGES_REQUESTED** [review]

File: source/modules/elements/elements.pug [hunk 15]
```diff
@@ -0,0 +1,147 @@
...
+//-
+//- Elements :: Textarea
+//-
+mixin textarea(label)
+  .form-group
+    if label !== ''
+      label= label
+    textarea(
+      class='form-control',
+      placeholder= 'Placeholder'
```
Comment (emily0604) [inline source/modules/elements/elements.pug]: Do we need this.

File: source/modules/sidebar/sidebar.pug [hunk 16]
```diff
@@ -0,0 +1,13 @@
+mixin sidebar(data)
+  .sidebar
+    h2.sidebar__user-avatar
+        a(href="#") User avatar image
```
Comment (emily0604) [inline source/modules/sidebar/sidebar.pug]: indent

File: source/sass/layout/_style-guide.scss [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+      &::before {
+        font-size: 2.25rem;
+        counter-increment: item;
+        content: counter(item, decimal-leading-zero) ' ';
+      }
+    }
+  }
+
+.guide-content__title {
+  @extend .title--black;
```
Comment (emily0604) [inline source/sass/layout/_style-guide.scss]: please make sure using extend for right purpose

File: source/elements.pug [hunk 18]
```diff
@@ -1,480 +0,0 @@
-include modules/heading/heading
-
-doctype html
```
Comment (emily0604) [inline source/elements.pug]: why remove this. this page is used for call all mixin we have

**Review by emily0604 — CHANGES_REQUESTED** [review]

**Review by emily0604 — APPROVED** [review]

**Pull request merged**